src/Controls/src/SourceGen/GeneratorHelpers.cs

```diff
@@ -53,11 +53,27 @@ public static string EscapeIdentifier(string identifier)
 		fileOptions.TryGetValue("build_metadata.additionalfiles.ManifestResourceName", out var manifestResourceName);
 		fileOptions.TryGetValue("build_metadata.additionalfiles.RelativePath", out var relativePath);
 		fileOptions.TryGetValue("build_property.targetframework", out var targetFramework);
-		fileOptions.TryGetValue("build_property._MauiForceXamlSourcegen", out var forceSourceGen);
 		fileOptions.TryGetValue("build_property.Configuration", out var configuration);
-		return new ProjectItem(additionalText, targetPath: targetPath, relativePath: relativePath, manifestResourceName: manifestResourceName, kind: kind, targetFramework: targetFramework)
+
+		fileOptions.TryGetValue("build_metadata.additionalfiles.Inflator", out var inflator);
+
+
+		var xamlinflator = string.IsNullOrEmpty(inflator) ? 0 : inflator!.Split(',').Select(s => s.Trim()).Select(s =>
+		{
+			if (!Enum.TryParse<XamlInflator>(s, true, out var xinfl))
+				throw new InvalidOperationException($"Invalid inflator '{s}' for {additionalText.Path}.");
+			return xinfl;
+		}).Aggregate(0, (acc, xamlinflator) => acc | (int)xamlinflator);
```

Comment:
There is a lot of System.Linq used here, does this run for every .xaml file? How often does it run per keystroke in Visual Studio?

You might be able to write the same thing with `foreach` and `|=` to just be a bit easier to read and faster.

Reply:
we could also add a layer of caching to avoid invalidation if metadata doesn't change